### DISCUSSION TITLE
Using both usb_host and usb_device

### DISCUSSION THREAD

**Discussion by ggasilveira** [hathach/tinyusb#1432] [Q&A] [answered]
Is it possible to use TinyUSB both as host and device at the same time?
Using Raspberry pi pico if it makes any difference.

I would like to use some pins on my board to communicate with a keyboard, and the board's usb to act as a device.

**Comment by hathach** [comment, marked as answer]
It is possible, I am actually working to integrate pio-usb as host, you could check out testing code here. Last time I run, it works but there could be some interim/missing piece of code etc..
 
https://github.com/hathach/tinyusb/tree/pio-host/examples/host/hid_to_cdc

There is still lots of work to get it running well, but we getting there.
